lightning/src/ln/channelmanager.rs

```diff
@@ -2569,6 +2569,13 @@ where
 	#[cfg(any(test, feature = "_test_utils"))]
 	pub(super) per_peer_state: FairRwLock<HashMap<PublicKey, Mutex<PeerState<SP>>>>,
 
+	/// We only support using one of [`ChannelMonitorUpdateStatus::InProgress`] and
+	/// [`ChannelMonitorUpdateStatus::Completed`] without restarting. Because the API does not
```

Comment:
Some thoughts: is there a way to enforce this with the compiler? And is it a goal of the project to continue supporting both sync + async persistence forever?

Reply:
> We definitely want to support both at the project level, though, for the same reason we want to support async and sync everywhere in the project - Rust-async-native projects are becoming more common, but Rust-non-async projects also exist (as well as platforms where that's required) as well as our current bindings are not async.

To clarify, by sync + async persistence I mean supporting either returning `ChannelMonitorUpdateStatus::Completed` or `::InProgress`. It feels like we expect most production users to persist data asynchronously and then later call `channel_monitor_updated`, so it could be reasonable to drop support for the `::Completed` option down the line. Seems like users that are the exception to that can just call `channel_monitor_updated` immediately, as well.

Reply:
Maybe, seems worth exploring though. Bookmarking to discuss further.

Reply:
A luxury API for a minority group of sync users may not be the best choice long term?

But if sync users would need to use async with an immediate callback now, I think it is no longer so clear that async is not production ready?

Hopefully we can get to super 100% soon.

Reply:
PR is not requiring that, but just following up on @valentinewallace comment. That if we would switch to async-interface-only now and just instruct sync users to call `channel_monitor_updated` immediately, it wouldn't be so clear on the API level what the experimental part still is.

Reply:
In general I think it's a good goal to have a minimal API surface, not saying this has to happen right now. Although async persist is experimental right now, you did state above that the same code paths are used in both places so hopefully that means it's getting close :p  

> Maybe? I imagine many small nodes will still want the simplicity of sync.

Re: this, ISTM those nodes will use LDK Node so they won't even know what the underlying LDK core API is here?